java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java

```diff
@@ -133,40 +142,135 @@ public void onCompleted() {
     }
 
     @Override
-    public BarrageTable entireTable() {
-        return partialTable(null, null, false);
+    public boolean isCompleted() {
+        return completed;
+    }
+
+    @Override
+    public synchronized void waitForCompletion() throws InterruptedException {
```

Comment:
I think we should make this method `private`. If we have a use-case for waiting for asynchronous subscriptions to complete, we should issue a future or let the caller pass a callback.